tools/osx-productsign.sh

```diff
@@ -1,12 +1,14 @@
-#!/bin/bash
+#!/bin/sh
 
 set -x
 set -e
 
-if [ "X$SIGN" == "X" ]; then
-  echo "No SIGN environment var.  Skipping codesign." >&2
+# shellcheck disable=SC2154
+[ -z "$SIGN" ] && \
+  echo "No SIGN environment var.  Skipping codesign." >&2 && \
   exit 0
-fi
 
+# shellcheck disable=SC2154
 productsign --sign "$SIGN" "$PKG" "$PKG"-SIGNED
+# shellcheck disable=SC2154
```

Comment:
Instead of these two disable directives, would it make sense to add a `-z "$PKG"` check similar to the one for `$SIGN` on line 7? Both are guaranteed by the Makefile so it would seem that we should either check for both or assume both, but we are checking for just one and assuming the other. Or is that outside the scope of this PR?